ui/widgets/dialog.js

```diff
@@ -437,7 +438,8 @@ $.widget( "ui.dialog", {
 			}
 		} );
 
-		uiDialogTitle = $( "<span>" ).uniqueId().prependTo( this.uiDialogTitlebar );
+		uiDialogTitle = $( this.options.uiDialogTitleTagName )
```

Comment:
Passing user input directly to the jQuery factory is dangerous - apart from obvious abuse when people pass selectors instead of tags here, they can e.g. pass `<script>`, etc. Protecting from that would require significantly more code, so I'm thinking we should actually change the approach.

How about introducing an option called `uiDialogTitleHeadingLevel` with the default of `0` and possible values all integers between `0` & `6`; you can use `Number.isInteger` for that. We can check the value and revert to `0` if it's invalid. Then, for `0` we use `<span>` but for other values we use a proper heading level.

That should cover the important use cases here and avoid possible security issues. Thoughts?

Reply:
in my initial draft i kept the <> out of the option but thought it would over complicate things so i ended up with the current version. but i agree that your suggested approach would be the better choice. would something like that work? 

```function between ( headingLvl ) {
  return Number.isInteger( headingLvl ) && headingLvl > 0 && headingLvl <= 6 ? "h" + headingLvl : "span";
};
uiDialogTitle = $( "<" + between( this.options.uiDialogTitleHeadingLevel ) + ">" ).uniqueId().prependTo( this.uiDialogTitlebar );```

i am not sure if it would be a clean code wise and i also dont know what the preferred name for the function would be. but the shown snippet is at least already working: the only difference to your suggestion is that the condition is only true between 1 and 6 everything else 0 included becomes a span. 

and i would also add a few test cases.

Reply:
That works, but since this logic is quite simple and only used in a single place, it's not necessary to create a function for it, I think. You can just define a local variable with the computed tag name & use it below.

Try to keep lines ideally below 80 characters (and definitely below 100), breaking them into multiple lines when necessary.

And yes, a few test cases would be needed, when the option was:
1. not passed (testing default behavior).
2. `0`
3. `1`
4. `6`
5. `7` (testing that it reverts to `span`)
6. `"foo"` (testing that it reverts to `span`)

Those should be enough, I think. Optionally, instead of testing `1` & `5` individually, you can create a loop through values 1-6, but I don't think it's absolutely necessary.

Reply:
ok thanks! the suggested changes are in place and the tests are green.